src/common/feedGenerator.ts

```diff
@@ -113,32 +113,65 @@ export const getExcludedAdvancedSettings = async (
   return excludedSettings.map((adv) => adv.id);
 };
 
-export const feedToFilters = async (
+const getExcludedSources = async (
   con: ORMConnection,
   feedId: string,
   userId: string,
+  enableSettingsExperiment: boolean,
+) => {
+  let query = con
+    .getRepository(Source)
+    .createQueryBuilder('s')
+    .select('s.id AS "id"');
+
+  if (enableSettingsExperiment) {
+    const settings = await getExcludedAdvancedSettings(con, feedId, userId);
+    query = query.where(
+      `s."advancedSettings" && ARRAY[:...settings]::integer[]`,
+      { settings },
+    );
+  } else {
+    query = query.where((qb) => {
+      const subQuery = qb
+        .subQuery()
+        .select('adv.id')
+        .from(AdvancedSettings, 'adv')
+        .leftJoin(
+          FeedAdvancedSettings,
+          'fas',
+          'adv.id = fas."advancedSettingsId" AND fas."feedId" = :feedId',
+          { feedId },
+        )
+        .where('COALESCE(fas.enabled, adv."defaultEnabledState") = false')
+        .getQuery();
+
+      return `s."advancedSettings" && array(${subQuery})`;
+    });
+  }
```

Comment:
I think it's better to use the same implementation but in one case take into consideration the flags and on the other not. It will make it more consistent and predictable. Easier to maintain as well

Reply:
I know that your original concern was performance but for now I prefer simplicity over performance. If we see that it still remains as a bottleneck we will fix it :)